asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+# - SQ: 44a, 44b
+# - Sample:
+# - Company_size_owner
+# - Company size_employee
+
+# %%
+# CompanySizeOwner V2
+# Create separate dataframe that stacks all answer columns
+df_44 = pd.DataFrame()
+for n in range(0, len(ans_44) - 1):
+
+    # Create new column for each answer option
+    response = ans_44[n]
+    new_col = "SQ44Ans"+str(n + 1)
+    data[new_col] = data.apply(transform_44, axis=1)
+
+    # Stack columns
+    df = data.filter(["CompanySizeOwnerV2", new_col])
+    df = df.rename(columns={new_col:"SQ44"})
+    df_44 = pd.concat([df_44, df])
+
+# Generate crosstab dataframe
+df_606_crosstab_owner_V2 = crosstable("CompanySizeOwnerV2",
+                                      df_44,
+                                     "SQ44",
+                                     ans_44
+                                     )
+df_606_crosstab_owner_V2 = df_606_crosstab_owner_V2.fillna(0)
+df_606_crosstab_owner_V2 = df_606_crosstab_owner_V2.astype(int)
+
+# Save to .csv and display
+df_606_crosstab_owner_V2.to_csv("../../outputs/section6/csv/recoded/606_owner_V2_recoded.csv")
+df_606_crosstab_owner_V2
+
+# %%
+# Generate figure
+groupedbar(df_606_crosstab_owner_V2,
+           "606_owner_V2_recoded",
+           "SQ 44a. Why do you only offer repairs, servicing and maintenance only for heat pump systems you have installed?")
+
+# %%
+# Generate figure
+donut(df_606_crosstab_owner_V2,
+           "606_owner_V2_recoded",
+           "SQ 44a. Why do you only offer repairs, servicing and maintenance only for heat pump systems you have installed?")
+
+# %%
+# CompanySizeEmployee V2
+# Create separate dataframe that stacks all answer columns
+df_44 = pd.DataFrame()
+for n in range(0, len(ans_44) -1 ):
+
+    # Create new column for each answer option
+    response = ans_44[n]
+    new_col = "SQ44Ans"+str(n + 1)
+    data[new_col] = data.apply(transform_44, axis=1)
+
+    # Stack columns
+    df = data.filter(["CompanySizeEmployeeV2", new_col])
+    df = df.rename(columns={new_col:"SQ44"})
+    df_44 = pd.concat([df_44, df])
+
+# Generate crosstab dataframe
+df_606_crosstab_employee_V2 = crosstable("CompanySizeEmployeeV2",
+                                         df_44,
+                                         "SQ44",
+                                         ans_44
+                                         )
+df_606_crosstab_employee_V2 = df_606_crosstab_employee_V2.fillna(0)
+df_606_crosstab_employee_V2 = df_606_crosstab_employee_V2.astype(int)
+
+# Save to .csv and display
+df_606_crosstab_employee_V2.to_csv("../../outputs/section6/csv/recoded/606_employee_V2_recoded.csv")
+df_606_crosstab_employee_V2
+
+# %%
+# Generate figure
+groupedbar(df_606_crosstab_employee_V2,
+           "606_employee_V2_recoded",
+           "SQ 44b. Why does the firm you work for offer repairs, servicing and maintenance only for heat pump systems you have installed?")
+
+# %%
+# Generate figure
+donut(df_606_crosstab_employee_V2,
```

Comment:
Currently in the donut plot, values with zero are showing e.g. "I only work on equipment I understand or have been trained on" with 0%.